### PR TITLE
#41 - Add fetching workflow jobs

### LINKED ISSUE #41: Add fetching workflow jobs
Create service and tests for fetching workflow jobs

### PR DESCRIPTION
**Pull request by orlowski11**
This should close #41

added services and tests for:

- fetching workflow jobs
- getting runner data from labels
- calculating job time

### CODE REVIEW

File: app/Services/FetchWorkflowJobsService.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        $repository = Repository::query()
+            ->where("id", $workflowRunDto->repositoryId)
+            ->firstOrFail();
+
+        $organization = Organization::query()
+            ->where("id", $repository->organization_id)
+            ->firstOrFail();
+
+        $user = User::query()->where("id", Auth::user()->id)->firstOrFail();
```
Comment (kamilpiech97) [inline app/Services/FetchWorkflowJobsService.php:43]: Is it needed?

Comment (orlowski11) [reply app/Services/FetchWorkflowJobsService.php:43]: I think so

File: app/Http/Integrations/Requests/GetWorkflowJobsRequest.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    public function resolveEndpoint(): string
+    {
+        $repository = Repository::query()
+            ->where("id", $this->workflowRunDto->repositoryId)
+            ->firstOrFail();
+
+        $organization = Organization::query()
+            ->where("id", $repository->organization_id)
+            ->firstOrFail();
```
Comment (kamilpiech97) [inline app/Http/Integrations/Requests/GetWorkflowJobsRequest.php]: Here, we can pass oragnization and repository ids as parameters.
`firstOrFail` actions we can do in controller ot smth like that.

File: app/Http/Integrations/Requests/GetWorkflowJobsRequest.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        return "/repos/" . $organization->name . "/" . $repository->name . "/actions/runs/" . $this->workflowRunDto->githubId . "/jobs";
+    }
+
+    public function createDtoFromResponse(Response $response): Collection
+    {
+        $workflowRun = WorkflowRun::query()->where("github_id", $this->workflowRunDto->githubId)->firstOrFail();
+        $getRunnerDataService = new GetRunnerDataService();
```
Comment (kamilpiech97) [inline app/Http/Integrations/Requests/GetWorkflowJobsRequest.php]: `fail` is possible?

File: app/Services/CalculateJobTimeService.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        $completedAt = new DateTime($completedAt);
+
+        $interval = $startedAt->diff($completedAt);
+
+        $seconds = ($interval->days * 24 * 60 * 60) + 
+                ($interval->h * 60 * 60) + 
+                ($interval->i * 60) + 
+                $interval->s;
+
+        $minutes = (int)ceil($seconds / 60);
```
Comment (kamilpiech97) [inline app/Services/CalculateJobTimeService.php]: Let's use [Carbon](https://carbon.nesbot.com/docs/), he has all of of methods for things like that

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: app/Http/Integrations/Requests/GetWorkflowJobsRequest.php (hunk 3 above)
Comment (orlowski11) [reply app/Http/Integrations/Requests/GetWorkflowJobsRequest.php]: shouldn't be, changed to `first()`

File: app/Http/Integrations/Requests/GetWorkflowJobsRequest.php (hunk 2 above)
Comment (orlowski11) [reply app/Http/Integrations/Requests/GetWorkflowJobsRequest.php]: now passing organization and repository name as parameters

File: app/Services/CalculateJobTimeService.php (hunk 4 above)
Comment (orlowski11) [reply app/Services/CalculateJobTimeService.php]: changed service to use Carbon

File: app/Services/CalculateJobTimeService.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+class CalculateJobTimeService
+{
+    public function calculate(string $startedAt, string $completedAt): int
+    {
+        $startedAt = Carbon::create($startedAt); 
+        $completedAt = Carbon::create($completedAt);
+
+        $diffInSeconds = $startedAt->diffInSeconds($completedAt);
+
+        $diffInMinutes = (int)ceil($diffInSeconds / 60);
```
Comment (kamilpiech97) [inline app/Services/CalculateJobTimeService.php:18]: Carbon should have method to diff in minutes with rounding value, I think so.

File: app/Services/FetchWorkflowJobsService.php (hunk 1 above)
Comment (kamilpiech97) [reply app/Services/FetchWorkflowJobsService.php:43]: `Auth::user()` method getting user data too, and in that line you are doing twice the same thing.

File: app/Services/CalculateJobTimeService.php (hunk 5 above)
Comment (orlowski11) [reply app/Services/CalculateJobTimeService.php:18]: `diffInMinutes()` rounds value, but here the value has to be always rounded **up**

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**